users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -618,13 +993,137 @@ def run_mel_audio_perturbation():
+        "increase_epochs": 180,
+        "decrease_epochs": 180,
+        "final_epochs": 0,
+    }
+
+    perturbation_args = [
+        {"codecs": [{"encoding": "ULAW", "prob": 0.3, "minimum": 250, "maximum": 260}]},
+        {"codecs": [{"encoding": "ULAW", "prob": 0.7, "minimum": 250, "maximum": 260}]},
+        {"codecs": [{"encoding": "ULAW", "prob": 0.7, "minimum": 240, "maximum": 270}]},
+        {"codecs": [{"encoding": "ULAW", "prob": 1, "minimum": 250, "maximum": 260}]},
+        {"tempo": {"prob": 0.3, "minimum": 0.83, "maximum": 1.17}},
+        {"tempo": {"prob": 0.8, "minimum": 0.83, "maximum": 1.17}},
+        {"tempo": {"prob": 0.3, "minimum": 0.7, "maximum": 1.3}},
+        {"tempo": {"prob": 0.8, "minimum": 0.7, "maximum": 1.3}},
+        {"tempo": {"prob": 1, "minimum": 0.83, "maximum": 1.17}},
+        {"tempo": {"prob": 1, "minimum": 0.7, "maximum": 1.3}},
+        {"speed": {"prob": 0.5, "minimum": 0.88, "maximum": 1.12}},
+        {"speed": {"prob": 0.3, "minimum": 0.88, "maximum": 1.12}},
+        {"speed": {"prob": 0.7, "minimum": 0.88, "maximum": 1.12}},
+        {"speed": {"prob": 0.6, "minimum": 0.8, "maximum": 1.2}},
+        {"speed": {"prob": 0.6, "minimum": 0.9, "maximum": 1.1}},
+        {"speed": {"prob": 1, "minimum": 0.9, "maximum": 1.1}},
+        {"speed": {"prob": 1, "minimum": 0.8, "maximum": 1.2}},
+        {"non_linearity": {"prob": 0.7, "minimum": 0.9, "maximum": 1.1}},
+        {"non_linearity": {"prob": 0.3, "minimum": 0.9, "maximum": 1.1}},
+        {"non_linearity": {"prob": 1, "minimum": 0.9, "maximum": 1.1}},
+        {"preemphasis": {"prob": 0.3, "minimum": 0.94, "maximum": 1.0, "default": 0.97}},
+        {"preemphasis": {"prob": 0.7, "minimum": 0.94, "maximum": 1.0, "default": 0.97}},
+        {"preemphasis": {"prob": 1.0, "minimum": 0.94, "maximum": 1.0, "default": 0.97}},
+        {"preemphasis": {"prob": 0.7, "minimum": 0.90, "maximum": 1.0, "default": 0.97}},
+        {"preemphasis": {"prob": 0.7, "minimum": 0.90, "maximum": 1.0, "default": 0.95}},
+    ]
+
+    nn_base_args = {}
+
+    for args in perturbation_args:
+        exp_name_suffix, report_args = args_to_key_and_report_strings(args)
+
+        # Construct the exp_name and report_args
+        exp_name = f"mel_bs2x5k_perturb_from_checkpoint_24_{exp_name_suffix}"
+        nn_base_args[exp_name] = dict(
+            returnn_args={
+                "extra_args": {
+                    "audio_perturb_args": args,
+                    "audio_perturb_runner": CodeWrapper("WaveformPerturbation(**audio_perturb_args)"),
+                    "conv_pad_seq_len_to_power": 1.5,
+                    "accum_grad_multiple_step": 2,
+                    "preload_from_files": {
+                        "existing-model": {
+                            "filename": nn_system.train_jobs["conformer_bs2x5k_lgm80_baseline"].out_checkpoints[24],
+                            "init_for_train": True,
+                        }
+                    },
```

Comment:
You could move this to the initial definition of `returnn_args` and replace it here with `**returnn_args["extra_args"]`. This would make it easier to see what is modified in the loop.